File: regex/CSharp/regex1.cs
using System;
using System.Text.RegularExpressions;

class Regex1
{
  static void Main()
  {
    Regex r = new Regex( "[.]txt$" );

    if ( r.Match( "file.txt" ).Success ) {
      Console.WriteLine( "File name matched" );
    } else {
      Console.WriteLine( "File name did not match" );
    }
  }
}

File: regex/Java/Regex1.java
import java.util.regex.*;

class Regex1
{
  public static void main( String args[] )
  {
    Pattern pat = Pattern.compile( "\\S*.txt$" );
    Matcher mat = pat.matcher( "file.txt" );
    if ( mat.matches() ) {
      System.out.println( "File matches" );
    } else {
      System.out.println( "File does not match" );
    }
  }
}
